frame/support/src/lib.rs

```diff
@@ -2580,3 +2627,13 @@ pub mod pallet_prelude {
 /// 	```
 /// * use the newest nightly possible.
 pub use frame_support_procedural::pallet;
+
+/// Contains macro stubs for all of the pallet:: macros
+pub mod pallet_macros {
+	pub use frame_support_procedural::{
+		call_index, compact, config, constant, disable_frame_system_supertrait_check, error, event,
+		extra_constants, generate_deposit, generate_storage_info, generate_store, genesis_build,
+		genesis_config, getter, hooks, inherent, origin, storage, storage_prefix, storage_version,
+		type_value, unbounded, validate_unsigned, weight, whitelist_storage,
+	};
+}
```

Comment:
This module contains exports of all of the macro stubs, and is included automatically as part of the `pallet_prelude`. This allows people to access the docs for the stubs and also makes them show up automatically for plugins like `rust-analyzer`.